src/Extension/FluentExtension.php

```diff
@@ -995,6 +999,18 @@ protected function detectLocalisedTableField($tables, $sql)
             }
         }
 
+        // Check sole "field" without table specifier ("name" without leading or trailing '.' and without quotes)
+        if (preg_match('/(?<![.])(?<field>\w+)(?![.])/i', $sql, $matches)) {
```

Comment:
Isn't this negative lookahead/behind pointless given the field should be quoted (ie. the field should always have a `"` delimiting it anyway)? 

Wouldn't this make more sense:

```
if (preg_match('/"?(?<field>\w+)"?(?!\.)/i', $sql, $matches)) {
```

Match a `"` maybe, then your field, then another `"` maybe, then not followed by a `.`.

Reply:
Actually I thought about this a little more. I'm pretty sure the negative lookahead you've got will only apply to the last character. So you'll have `Foo.Bar` matching and `$matches['field']` will be `Ba`.

I think you want a regex more like this:

```
/(?!\w+"?\.)(\w+)/i
```

This is a negative match ahead on anything followed by a dot, and then a match on whatever is left. All these regexes only work if the `$sql` being matched is only the field btw. If there's `SELECT` or `WHERE` conditions then it's going to be weird.

Reply:
Ya that's what I'm worried about actually... this method is used for more than just sort columns, I wonder if perhaps it would be better to check for quotes and add them if they're not present prior to this method being called when updating the sort.

Reply:
I wonder if something like this makes more sense:

```diff
@@ -376,6 +376,11 @@ class FluentExtension extends DataExtension
         // Iterate through each order clause, replacing each with the translated version
         $order = $query->getOrderBy();
         foreach ($order as $column => $direction) {
+            // Fix conditions where the column is a non-fully-qualified name and is not quoted
+            if (strpos($column, '.') === false && strpos($column, '"') === false) {
+                $column = '"' . $column . '"';
+            }
+            
             // Parse fragment for localised field and table
             list ($table, $field, $fqn) = $this->detectLocalisedTableField($tables, $column);
             if ($table && $field) {

```